src/software/thunderscope/field/field.py

```diff
@@ -21,6 +21,14 @@ def __init__(self):
         # Fields
         self.layers = []
 
+    def keyPressEvent(self, event):
+        for layer in self.layers:
+            layer.keyPressEvent(event)
+
+    def keyReleaseEvent(self, event):
+        for layer in self.layers:
+            layer.keyReleaseEvent(event)
+
```

Comment:
it might be good to add a register function here

```
def register_key_press_callback(self, callback):
       self.key_press_callbacks.append(callback)
```

and 

```
def register_key_release_callback(self, callback):
    self.key_release_callbacks.append(callback)
```

and then in `keyPress` and `keyRelease` even functions, you would need to add

```
for callback in self.key_press_callback:
     callback(event)
```

This would allow us to register other actions on keypresses on the field

Reply:
Do we want this done in world_layer instead of field? I'm thinking I would need to define the register functions in world_layer.py since the callback functions will need access to the protos in world_layer